packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx

```diff
@@ -114,24 +114,26 @@ export class TooltipHost extends BaseComponent<ITooltipHostProps, ITooltipHostSt
       }
     }
 
-    this.setState({
-      isTooltipVisible: true
-    });
+    this.toggleTooltip(true);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipMouseLeave(ev: any) {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipCallOutDismiss() {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
+  }
+
+  private toggleTooltip(isTooltipVisible: boolean) {
+    if (this.props.onTooltipToggle && this.state.isTooltipVisible !== isTooltipVisible) {
+      this.props.onTooltipToggle(isTooltipVisible);
+    }
```

Comment:
Hmm, I agree that its not a major issue but using callback will make it more consistent especially when we are comparing `this.state.isTooltipVisible !== isTooltipVisible`

I'm too worried about the following flow:
1. `toggleTooltip(true)` 
    - called `onTooltipToggle(true)`,
    - called `this.setState({isTooltipVisible: true})`

2. `toggleTooltip(false)` 
    - can potentially skip `onTooltipToggle()` call if last `setState(...)` is not applied yet
    - called `this.setState({isTooltipVisible: true})`